### PR TITLE
fix more issues

### PR DESCRIPTION
**Pull request by Etienne-Buschong**
- fixes type detail infinite recursion (i also support for function types in the same step)
- uses correct plugin name
- tries other method where a smaller ts-project is build on each seal hook, to reduce creation time of ts-project. I need to test/time this yet.

### CODE REVIEW

File: src/webpack-angular-types-plugin/plugin.ts [hunk 1]
```diff
@@ -12,27 +13,35 @@ import { generateClassInformation } from "./type-extraction/type-extraction";
 
 export class WebpackAngularTypesPlugin {
     apply(compiler: Compiler) {
-        compiler.hooks.compilation.tap("TestPlugin", (compilation) => {
+        const tsProject = new Project({
+            tsConfigFilePath: DEFAULT_TS_CONFIG_PATH,
+            skipLoadingLibFiles: true,
+            skipFileDependencyResolution: true,
+        });
```
Comment (kroeder) [inline src/webpack-angular-types-plugin/plugin.ts:20]: Can this be removed in favor of line 32?

**Review by kroeder — APPROVED** [review]
LGTM

File: src/webpack-angular-types-plugin/plugin.ts (hunk 1 above)
Comment (Etienne-Buschong) [reply src/webpack-angular-types-plugin/plugin.ts:20]: No this can not be removed. I currently use this tsProject to check which files from the changed modules of webpack (in the collectModulesToProcess) should be actually crawled for classes.

If we can somehow determine without this, which files are eligible for processing, then this can be removed.

TLDR: If moduleToProcess can be created without using the tsProject, then we can remove this.
